developer/src/kmc-kmn/src/compiler/compiler.ts

```diff
@@ -107,4 +111,72 @@ export class Compiler {
       return false;
     }
   }
-}
\ No newline at end of file
+
+  /**
+   *
+   * @param pattern UnicodeSet pattern such as `[a-z]`
+   * @param bufferSize guess as to the buffer size
+   * @returns UnicodeSet accessor object
+   */
+  public async parseUnicodeSet(pattern: string, bufferSize: number) : Promise<UnicodeSet> {
+    if (!bufferSize) {
+      bufferSize = 100;
+    }
+
+    const initOk = await this.init();
+    if (!initOk) {
+      throw Error(`WASM machinery didn't start up properly`);
+    }
+    // Module seems to be the usual name for it
+    const Module = this.wasmModule;
+    const buf = Module.asm.malloc(bufferSize * 2 * Module.HEAPU32.BYTES_PER_ELEMENT);
+    const rc = this._parseUnicodeSet(pattern, buf, bufferSize);
+    if (rc < 0) {
+      throw new UnicodeSetError(rc);
+    } else { // rc ≥0
+      const ranges = [];
+      const startu = (buf / Module.HEAPU32.BYTES_PER_ELEMENT);
+      for (let i = 0; i < rc; i++) {
+        const low = Module.HEAPU32[startu + (i * 2) + 0];
+        const high = Module.HEAPU32[startu + (i * 2) + 1];
+        ranges.push([low, high]);
+      }
+      return new UnicodeSet(pattern, ranges);
+    }
+  }
+}
+
+export class UnicodeSetError extends Error {
+  code: number;
+  constructor(code:number) {
+    super();
+    this.code = code;
+    this.message = `UnicodeSet error: ${code}`;
+  }
+}
+
+// from kmcmplib.h
+export const KMCMP_USET_OK = 0;
+export const KMCMP_ERROR_SYNTAX_ERR = -1;
+export const KMCMP_ERROR_HAS_STRINGS = -2;
+export const KMCMP_ERROR_UNSUPPORTED_PROPERTY = -3;
+export const KMCMP_ERROR_UNSUPPORTED = -4;
+export const KMCMP_FATAL_OUT_OF_RANGE = -5;
```

Comment:
We should translate these to kmc error values in messages.ts.

Reply:
+1